windows application build failed

### Did you check the documentation and it did not help?

- [x] I have read the [README](https://github.com/natsuk4ze/gal/blob/main/README.md)
- [x] I have searched the [existing documentation](https://github.com/natsuk4ze/gal/wiki)
- [x] I have searched the [existing answer](https://github.com/natsuk4ze/gal/discussions)
- [x] I have read the [contribution guide](https://github.com/natsuk4ze/gal/blob/main/CONTRIBUTING.md)

### Overview

Failure when running or compiling Windows applications

### Steps to reproduce

1.  Create a new flutter project
2.  Add gal to the project
3.  run `flutter build winodw`

### Code sample

<details><summary>Code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  // This widget is the root of your application.
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        colorScheme: ColorScheme.fromSeed(seedColor: Colors.deepPurple),
      ),
      home: const MyHomePage(title: 'Flutter Demo Home Page'),
    );
  }
}

class MyHomePage extends StatefulWidget {
  const MyHomePage({super.key, required this.title});

  final String title;

  @override
  State<MyHomePage> createState() => _MyHomePageState();
}

class _MyHomePageState extends State<MyHomePage> {
  int _counter = 0;

  void _incrementCounter() {
    setState(() {
      _counter++;
    });
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        backgroundColor: Theme.of(context).colorScheme.inversePrimary,
        title: Text(widget.title),
      ),
      body: Center(
        child: Column(
          mainAxisAlignment: MainAxisAlignment.center,
          children: <Widget>[
            const Text('You have pushed the button this many times:'),
            Text(
              '$_counter',
              style: Theme.of(context).textTheme.headlineMedium,
            ),
          ],
        ),
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: _incrementCounter,
        tooltip: 'Increment',
        child: const Icon(Icons.add),
      ), // This trailing comma makes auto-formatting nicer for build methods.
    );
  }
}

```

</details>


### Environment

<details><summary>Logs</summary>

```console
[!] Flutter (Channel stable, 3.32.6, on Microsoft Windows [Version 10.0.26100.4652], locale zh-CN) [359ms]
    • Flutter version 3.32.6 on channel stable at C:\pro\flutter
    ! Warning: `dart` on your path resolves to C:\pro\dart-sdk\bin\dart.exe, which is not inside your current Flutter
      SDK checkout at C:\pro\flutter. Consider adding C:\pro\flutter\bin to the front of your path.
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 077b4a4ce1 (8 days ago), 2025-07-08 13:31:08 -0700
    • Engine revision 72f2b18bb0
    • Dart version 3.8.1
    • DevTools version 2.45.1
    • If those were intentional, you can disregard the above warnings; however it is recommended to use "git" directly
      to perform update checks and upgrades.

[✓] Windows Version (11 Pro 64-bit, 24H2, 2009) [957ms]

[✓] Android toolchain - develop for Android devices (Android SDK version 35.0.0) [6.2s]
    • Android SDK at C:\Users\nil\AppData\Local\Android\sdk
    • Platform android-35, build-tools 35.0.0
    • Java binary at: C:\Program Files\Android\Android Studio1\jbr\bin\java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.6+-13391695-b895.109)
    • All Android licenses accepted.

[✓] Chrome - develop for the web [80ms]
    • Chrome at C:\Program Files\Google\Chrome\Application\chrome.exe

[✓] Visual Studio - develop Windows apps (Visual Studio Community 2022 17.14.9 (July 2025)) [80ms]
    • Visual Studio at C:\Program Files\Microsoft Visual Studio\2022\Community
    • Visual Studio Community 2022 version 17.14.36310.24
    • Windows 10 SDK version 10.0.26100.0

[!] Android Studio (version 2023.1) [15ms]
    • Android Studio at C:\Program Files\Android\Android Studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    ✗ Unable to determine bundled Java version.
    • Try updating or re-installing Android Studio.

[✓] Android Studio (version 2025.1.1) [14ms]
    • Android Studio at C:\Program Files\Android\Android Studio1
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 21.0.6+-13391695-b895.109)

[✓] VS Code (version 1.102.0) [14ms]
    • VS Code at C:\Users\nil\AppData\Local\Programs\Microsoft VS Code
    • Flutter extension version 3.114.0

[✓] Connected device (3 available) [208ms]
    • Windows (desktop) • windows • windows-x64    • Microsoft Windows [Version 10.0.26100.4652]
    • Chrome (web)      • chrome  • web-javascript • Google Chrome 138.0.7204.101
    • Edge (web)        • edge    • web-javascript • Microsoft Edge 138.0.3351.83

[✓] Network resources [2.1s]
    • All expected network resources are available.
```

</details>


### Logs

<details><summary>Logs</summary>

```console
gal_plugin.obj : error LNK2019: unresolved external symbol WINRT_IMPL_RoGetActivationFactory referenced in function "struct winrt::hresult __cdecl winrt::impl::get_runtime_activation_factory_impl<0>(struct winrt::param::hstring const &,struct winrt::guid const &,void * *)" (??$get_runtime_activation_factory_impl@$0A@@impl@winrt@@YA?AUhresult@1@AEBUhstring@param@1@AEBUguid@1@PEAPEAX@Z) [C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\gal_plugin.vcxproj]
gal_plugin.obj : error LNK2019: unresolved external symbol WINRT_IMPL_RoOriginateLanguageException referenced in function "private: void __cdecl winrt::hresult_error::originate(struct winrt::hresult,void *,struct winrt::impl::slim_source_location const &)" (?originate@hresult_error@winrt@@AEAAXUhresult@2@PEAXAEBUslim_source_location@impl@2@@Z) [C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\gal_plugin.vcxproj]
gal_plugin.obj : error LNK2019: unresolved external symbol WINRT_IMPL_RoFailFastWithErrorContext referenced in function "void __cdecl winrt::terminate(void)" (?terminate@winrt@@YAXXZ) [C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\gal_plugin.vcxproj]
gal_plugin.obj : error LNK2019: unresolved external symbol WINRT_IMPL_RoTransformError referenced in function "bool __cdecl winrt::impl::report_failed_invoke(void)" (?report_failed_invoke@impl@winrt@@YA_NXZ) [C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\gal_plugin.vcxproj]
C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\Release\gal_plugin.dll : fatal error LNK1120: 4 unresolved externals [C:\Users\ooo\Desktop\New folder\hi\build\windows\x64\plugins\gal\gal_plugin.vcxproj]
```

</details>

--- a/windows/CMakeLists.txt
+++ b/windows/CMakeLists.txt
@@ -54,7 +54,7 @@ target_compile_definitions(${PLUGIN_NAME} PRIVATE FLUTTER_PLUGIN_IMPL)
 # dependencies here.
 target_include_directories(${PLUGIN_NAME} INTERFACE
   "${CMAKE_CURRENT_SOURCE_DIR}/include")
-target_link_libraries(${PLUGIN_NAME} PRIVATE flutter flutter_wrapper_plugin)
+target_link_libraries(${PLUGIN_NAME} PRIVATE flutter flutter_wrapper_plugin windowsapp)
 
 # List of absolute paths to libraries that should be bundled with the plugin.
 # This list could contain prebuilt libraries, or libraries created by an

